File: app/src/main/java/de/sneakpeek/data/MovieInfoCast.java
package de.sneakpeek.data;


import android.os.Parcel;
import android.os.Parcelable;

import com.squareup.moshi.Json;

public class MovieInfoCast implements Parcelable {

    public final static Parcelable.Creator<MovieInfoCast> CREATOR = new Creator<MovieInfoCast>() {

        public MovieInfoCast createFromParcel(Parcel in) {
            MovieInfoCast instance = new MovieInfoCast();
            instance.character = ((String) in.readValue((String.class.getClassLoader())));
            instance.name = ((String) in.readValue((String.class.getClassLoader())));
            return instance;
        }

        public MovieInfoCast[] newArray(int size) {
            return (new MovieInfoCast[size]);
        }

    };

    @Json(name = "character") public String character;
    @Json(name = "name") public String name;

    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(character);
        dest.writeValue(name);
    }

    public int describeContents() {
        return 0;
    }
}

File: app/src/main/java/de/sneakpeek/data/MovieInfoCast.kt
package de.sneakpeek.data


import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class MovieInfoCast(@Expose @SerializedName("character") var character: String?,
                         @Expose @SerializedName("name") var name: String?) : Parcelable {

    companion object {
        @JvmField val CREATOR: Parcelable.Creator<MovieInfoCast> = object : Parcelable.Creator<MovieInfoCast> {
            override fun createFromParcel(source: Parcel): MovieInfoCast = MovieInfoCast(source)
            override fun newArray(size: Int): Array<MovieInfoCast?> = arrayOfNulls(size)
        }
    }

    constructor(source: Parcel) : this(
            source.readString(),
            source.readString()
    )

    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(character)
        dest.writeString(name)
    }
}


